project/dependencies.py

```
@@ -1,83 +1,87 @@
+
+    async with httpx.AsyncClient() as client:
+        try:
+            r = await client.get(jwks_url)
+            r.raise_for_status()
+        except HTTPError:
+            logger.exception("Failed to read OIDC config")
+
+            raise HTTPException(
+                status_code=status.HTTP_502_BAD_GATEWAY,
+                detail="Auth provider is unavailable",
+            )
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)